hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -735,6 +735,15 @@ object DataSourceWriteOptions {
     .withDocumentation("Controls whether overwrite use dynamic or static mode, if not configured, " +
       "respect spark.sql.sources.partitionOverwriteMode")
 
+  val DELETE_RECORD_SCHEMA_PRUNING: ConfigProperty[String] = ConfigProperty
+    .key("hoodie.datasource.delete.record.schema.pruning.enabled")
+    .defaultValue("true")
```

Comment:
Can we eliminate this option because it looks like a pure optimization.

Reply:
> Can we eliminate this option because it looks like a pure optimization.

Of course we can. Done~